core/src/docs/upgrade.md

```diff
@@ -1,3 +1,20 @@
+# Upgrade to v0.33
+
+## Public API
+
+OpenDAL 0.33 has redesigned the `Writer` API, replacing all instances of `writer.append()` with `writer.write()`. For more information, please refer to [`Writer`](crate::Writer).
```

Comment:
Does the merging have any impact on the original 'writer.write'?
It is necessary to remind the user.

Reply:
> Does the merging have any impact on the original 'writer.write'?

No, the behavior not change.

Reply:
So it does sound `append` a bit confusing.

Reply:
> So it does sound `append` a bit confusing.

Yep :rofl: